packages/catalog-generator/src/main/java/io/kaoto/camelcatalog/generator/CamelCatalogProcessor.java

```diff
@@ -0,0 +1,635 @@
+    private static final String SET_VARIABLES_DEFINITION = "org.apache.camel.model.SetVariablesDefinition";
+    private final ObjectMapper jsonMapper;
+    private final CamelCatalog camelCatalog;
+    private final CamelYamlDslSchemaProcessor schemaProcessor;
+    private boolean verbose;
+
+    public CamelCatalogProcessor(CamelCatalog camelCatalog, ObjectMapper jsonMapper,
+            CamelYamlDslSchemaProcessor schemaProcessor, boolean verbose) {
+        this.jsonMapper = jsonMapper;
+        this.camelCatalog = camelCatalog;
+        this.schemaProcessor = schemaProcessor;
+        this.verbose = verbose;
+    }
+
+    /**
+     * Create Camel catalogs customized for Kaoto usage.
+     *
+     * @return
+     */
+    public Map<String, String> processCatalog() throws Exception {
+        var answer = new LinkedHashMap<String, String>();
+        var componentCatalog = getComponentCatalog();
+        var dataFormatCatalog = getDataFormatCatalog();
+        var languageCatalog = getLanguageCatalog();
+        var modelCatalog = getModelCatalog();
+        var patternCatalog = getPatternCatalog();
+        var entityCatalog = getEntityCatalog();
+        var loadBalancerCatalog = getLoadBalancerCatalog();
+        answer.put("components", componentCatalog);
+        answer.put("dataformats", dataFormatCatalog);
+        answer.put("languages", languageCatalog);
+        answer.put("models", modelCatalog);
+        answer.put("patterns", patternCatalog);
+        answer.put("entities", entityCatalog);
+        answer.put("loadbalancers", loadBalancerCatalog);
+        return answer;
+    }
+
+    /**
+     * Get aggregated Camel component Catalog.
+     *
+     * @return
+     * @throws Exception
+     */
+    public String getComponentCatalog() throws Exception {
+        var answer = jsonMapper.createObjectNode();
+        camelCatalog.findComponentNames().stream().filter(component -> !component.isEmpty()).sorted()
+                .forEach((name) -> {
+                    try {
+                        var model = (ComponentModel) camelCatalog.model(Kind.component, name);
+                        var json = JsonMapper.asJsonObject(model).toJson();
+                        var catalogNode = (ObjectNode) jsonMapper.readTree(json);
+                        generatePropertiesSchema(catalogNode);
+                        answer.set(name, catalogNode);
+                    } catch (Exception e) {
+                        if (verbose) {
+                            LOGGER.warning("The component definition for " + name + " is null.\n" + e.getMessage());
+                        }
+                    }
+                });
+        StringWriter writer = new StringWriter();
+        var jsonGenerator = new JsonFactory().createGenerator(writer).useDefaultPrettyPrinter();
+        jsonMapper.writeTree(jsonGenerator, answer);
+        return writer.toString();
+    }
+
+    private void generatePropertiesSchema(ObjectNode parent) throws Exception {
```

Comment:
Remove the declaration of thrown exception 'java.lang.Exception', as it cannot be thrown from method's body.